daisy/step_create_images_test.go

```diff
@@ -115,3 +116,84 @@ func TestCreateImagesRun(t *testing.T) {
 		}
 	}
 }
+
+func TestImageUsesAlphaFeaturesTrue(t *testing.T) {
+	tests := []struct {
+		desc      string
+		cia       []*ImageAlpha
+		wantResult bool
+	}{
+		{
+			desc: "Use AlphaAPI due to RolloutOverride.",
+			cia: []*ImageAlpha{
+				{
+					Image: computeAlpha.Image{
+						RolloutOverride: &computeAlpha.RolloutPolicy{
+							DefaultRolloutTime: time.Now().Format(time.RFC3339),
+							LocationRolloutPolicies: map[string]string{
+								"zones/one": time.Now().Add(time.Hour).Format(time.RFC3339),
+							},
+						},
+					},
+				},
+			},
+			wantResult: true,
+		},
+		{
+			desc: "Use AlphaAPI due to Deprecated.StateOverride.",
+			cia: []*ImageAlpha{
+				{
+					Image: computeAlpha.Image{
+						Deprecated: &computeAlpha.DeprecationStatus{
+							State: "OBSOLETE",
+							StateOverride: &computeAlpha.RolloutPolicy{
+								DefaultRolloutTime: time.Now().Format(time.RFC3339),
+								LocationRolloutPolicies: map[string]string{
+									"zones/one": time.Now().Add(time.Hour).Format(time.RFC3339),
+								},
+							},
+						},
+					},
+				},
+			},
+			wantResult: true,
+		},
+		{
+			desc: "Don't AlphaAPI, empty",
+			cia: []*ImageAlpha{{Image: computeAlpha.Image{}}},
```

Comment:
This will test the `imageAlpha.RolloutOverride = nil` but we should also test when `len(imageAlpha.RolloutOverride.DefaultRolloutTime) = 0` and:
* `imageAlpha.Deprecated == nil` or
* `imageAlpha.Deprecated.StateOverride == nil`

Reply:
Updated in [99c3847](https://github.com/GoogleCloudPlatform/compute-image-tools/pull/1608/commits/99c3847ba83eaf50d1fc32d551bd5343d185b45e)

- len(imageAlpha.RolloutOverride.DefaultRolloutTime) = 0 
  - I believe this is covered by `Do not use AlphaAPI, RolloutOverride present with all default values.` as `RolloutOverride: &computeAlpha.RolloutPolicy{}` initializes the struct and DefaultRolloutTime is a string which has a default value of empty string.

- imageAlpha.Deprecated == nil 
  - Added `Do not use AlphaAPI, Deprecated is nil.` I had thought this would be covered by `Do not use AlphaAPI, empty` since it's not defined there.

- imageAlpha.Deprecated.StateOverride == nil
  -  I added `Do not use AlphaAPI, deprecated with StateOverride = nil.`. I had thought `Do not use AlphaAPI, deprecated but with no StateOverride.` would have covers this as StateOverride should be nil since it wasn't defined and it's a pointer.